packages/workspace/src/browser/workspace-service.ts

```diff
@@ -65,13 +65,18 @@ export class WorkspaceService implements FrontendApplicationContribution {
     @inject(PreferenceSchemaProvider)
     protected readonly schemaProvider: PreferenceSchemaProvider;
 
+    @inject(LabelProvider)
+    protected readonly labelProvider: LabelProvider;
+
     protected applicationName: string;
+    protected workspaceName: string;
```

Comment:
Please don’t cache it, just compute on each call. Workspace can be changed and then you have to keep it in sync. Also does not need to be computed here, but in navigator-model

Reply:
sure I'll make the changes :)